### DISCUSSION TITLE
How is the certificate/key thing working?

### DISCUSSION THREAD

**Discussion by andih677** [mrbungle64/ioBroker.ecovacs-deebot#491] [Q&A]
Hello,
I used the ecovacs-deebot-adapter successfully the last 1,5 years. Now my Deebot 950 stops working because of the "clean up the drop sensors" problem. (The sensors are clean! I think its part of the planned obsolescence)

So I setup the Bumper-Server, hoping to get the robot working again.
Bumper runs and the robot is connected.
But I have the same problem with the adapter, logging "Unable to verifiy the first certificate", as described in 
[https://github.com/mrbungle64/ioBroker.ecovacs-deebot/issues/126](https://github.com/mrbungle64/ioBroker.ecovacs-deebot/issues/126)

I set up a new internal CA, issued the bumper server cert and started bumper with that. Copied the root-ca-cert to iobroker and added it  with
`[Service]
Environment="NODE_EXTRA_CA_CERTS=/path/to/cert/mycert.crt"`
(or do I need to add the bumper server's certificate here?)

I'm wondering about the file key.pem in `/iobroker/node_modules/ecovacs-deebot`
-- do I have to use the default key.pem that comes with the adapter? 
Tried to use a special client certificate issued by my new internal CA as key.pem - "unable to verify..."
Tried the public key from bumper server's cert as key.pem - "unable to verify..."
Tried the bumper server's cert as key.pem - "unable to verify..."

Whats the purpose of the key.pem-file and where did you get it? It says as name only "Default company"...
What can I try next?

Andy

**Comment by mrbungle64** [comment]
@andih677 

> Whats the purpose of the key.pem-file and where did you get it? It says as name only "Default company"...

To be honest, I can't give you much information about it...
It was already part of the ([sucks.js](https://github.com/joostth/sucks.js)) library before I forked it 😉 

Have you already checked [this doc](https://bumper.readthedocs.io/en/latest/Create_Certs/)?

**Comment by andih677** [comment]
Hi mrbungle64,
read it. 
I tried it with own certificates created with the xca tool. Saved the CA and bumper cert on bumper server (and if I access the bumper server my certificate shows up in the browser).
On client side there is only this short key.pem file but no key in it, so maybe it has nothing to do?

Still I'm not sure if ioBroker has correct access to the installed CA certificate to verify the bumper server's certificate.
Saved it in /usr/local/share/ca-certificates/extra  und did a "update-ca-certificates" so it gets the link in /etc/ssl/certs.
And after that I edited the systemctl for ioBroker to use the extra-path for the ca-certificate.

But I'm not in hurry since my robot stopped working caused by the vendor's programmed anti-drop sensors shutdown.  :-(
